cmd/check_container.go

```diff
@@ -186,15 +186,51 @@ var checkContainerCmd = &cobra.Command{
 				)
 			}
 
+			certImage, err := os.Open(path.Join(artifacts.Path(), certification.DefaultCertImageFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultCertImageFilename,
+				)
+			}
+			preflightResults, err := os.Open(path.Join(artifacts.Path(), certification.DefaultTestResultsFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultTestResultsFilename,
+				)
+			}
+			rpmManifest, err := os.Open(path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					certification.DefaultRPMManifestFilename,
+				)
+			}
+			logfile, err := os.Open(viper.GetString("logfile"))
+			if err != nil {
+				return fmt.Errorf(
+					"%w: could not open file for submission: %s",
+					err,
+					viper.GetString("logfile"),
+				)
+			}
 			submission.
 				// The engine writes the certified image config to disk in a Pyxis-specific format.
-				WithCertImageFromFile(path.Join(artifacts.Path(), certification.DefaultCertImageFilename)).
+				WithCertImage(certImage).
 				// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
-				WithPreflightResultsFromFile(path.Join(artifacts.Path(), certification.DefaultTestResultsFilename)).
+				WithPreflightResults(preflightResults).
 				// The certification engine writes the rpmManifest for images not based on scratch.
-				WithRPMManifestFromFile(path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename)).
+				WithRPMManifest(rpmManifest).
 				// Include the preflight execution log file.
-				WithArtifactFromFile(viper.GetString("logfile"))
+				WithArtifact(logfile)
+
+			if len(submission.Artifacts) > 0 {
+				submission.Artifacts[len(submission.Artifacts)-1].Filename = viper.GetString("logfile")
```

Comment:
This is going to be really confusing to remember in the future. Part of the logic is building all the attributes of the `Artifacts` struct (size, projectId, contentType, etc), then here we are going to add the filename to that struct? No one will remember that's how this works after this PR is merged. Can't we just have the filename be set in the builder?

Reply:
That's how I originally had it. I'm open to adding it back.